src/nanorc/node.py

```diff
@@ -507,9 +507,26 @@ def _on_enter_callback(self, event):
                 done = []
                 for child_node in appset:
                     if not child_node.included: continue
-
-                    if not child_node.sup.desc.proc.is_alive() or not child_node.sup.commander.ping():
+                    IsAlive = child_node.sup.desc.proc.is_alive()
+                    if not IsAlive:
+                        failed.append(child_node.name)
```

Comment:
`failed` is currently a list. For tracking purposes, it would be useful to move to a dictionary key-d on the failure mode:
`failed = { f:[] for f in ('dead', 'noping', 'error') }`
This makes easier print an error summary at the end of the timeout loop

Reply:
I'm not sure I completely understand what you mean by this